keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -3,703 +3,742 @@
+    | - name: cosi-bucket                             |
+    |   projected:                                    |
+    |     sources:                                    |
+    |     - secret:                                   |
+    |       name: bucketcreds1                        |
+    |-------------------------------------------------|
+```
 
-+ The admin creates a cluster-scoped `Bucket` representing the backend bucket.
-+ The user creates a `BucketRequest` referencing this `Bucket` instance and COSI follows the "Create Bucket" workflow.
-+ The user creates a `BucketAccessRequest` (BAR) which references their BR.
-+ COSI central controller sees a new `BucketAccessRequest` (BAR) and gets the `BAR.BucketAccessClass` (BAC).
-+ COSI creates the cluster-scoped `BucketAccess` (BA) filling in the field from the BAR and BAC.
-+ The sidecar sees the new BA and gRPC calls the driver to grant access to the backend bucket.
-+ COSI updates the BAR and BA to reference each other.
-+ The cosi-node-adapter, which has been waiting for the BAR to be processed, writes access tokens to the app pod's specified mount point.
+The volume `mountPath` will be the directory where bucket credentials and other related information will be served. 
 
-## Revoke Bucket Access
-This workflow describes the automation supporting revoking access to an existing backend bucket, and the deletion of the cluster-scoped `BucketAccess` instance.
+NOTE: the contents of the files served in mountPath will be a COSI generated file containing credentials and other information required for accessing the bucket. This is not intended to specify a mountpoint containing the bucket's data.
 
-Here is the workflow:
+###### 2. COSI Node Adapter serves BucketInfo in the specified directory
 
-+ COSI central controller sees the `BucketAccessRequest` (BAR) has been deleted.
-+ COSI gets the `BAR.BucketAccess` (BA).
-+ COSI deletes the BA, but finalizers prevent it from being garbage collected.
-+ The sidecar sees the BA has been deleted and gRPC calls the driver to revoke access to the backend bucket.
-+ COSI removes the BAR and BA finalizers and these instances are garbage collected.
+The above volume definition will prompt kubernetes to retrieve the secret and place it in the volumeMount path defined above. The contents of the secret will be of the format shown below:
 
-In addition, the cosi-node-adapter sees the app pod is terminating:
-+ cosi-node-adapter receives _NodeUnstageVolume_ request from kubelet.
-+ adapter removes its finalizer from both the BAR.BA and the BAR.BR.
+```
+    bucket.json
+    |-----------------------------------------------|
+    | {                                             |
+    |   apiVersion: "v1alpha1"                      |
+    |   "S3": {                                     |
+    |     "bucketName": "br-$uuid",                 |
+    |     "endpoint": "https://s3.amazonaws.com",   |
+    |     "accessKeyID": "AKIAIOSFODNN7EXAMPLE",    |
```

Comment:
Is auth part of the per-protocol specification (e.g. S3 _always_ puts it in this header and GCS always puts it in that one) or can two implementations of S3 handle auth differently?

IOW - are we asserting that auth-protocol and data-protocol are hard-coupled?

Reply:
Yes, they are hard coupled.

Reply:
I'm not sure I understand what that means.  I guess I am asking: For a protocol like S3, is the authn _always_ handled the exact same way, or does it depend on the provider?

Reply:
It is always handled the same way. An application will always use the S3 protocol which includes authn, regardless of the provider.

The OSP's admin APIs for user management is what I meant will be different. AWS IAM API looks different from MinIO's User management API.

Reply:
OK, thanks.